src/vm.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'config'
+require 'libvirt'
+require 'metal_log'
+
+module Metalware
+  class Vm
+    def initialize(libvirt_host, node, *args)
```

Comment:
Also, for whichever arguments we do still need it could be worth adding `attr_reader`s to this class. This tends to be a good idea to use for instance variables as it means you need to change less if things switch to/from instance variables or methods. It also gives you protection from accessing an invalid/typo-ed instance variable - accessing `@invalid` just returns `nil`, but accessing `self.invalid` will give an error which should make it obvious what you've missed.